test/integration/fixtures/pending/skip-sync-afterEach.fixture.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+describe('skip in afterEach', function () {
+  it('should run this test-1', function () {});
+  it('should run this test-2', function () {});
+
+  afterEach('should print DeprecationWarning', function () {
+    this.skip();
+    throw new Error('never throws this error');
```

Comment:
some sort of implicit return is definitely _not_ what I would expect.

Reply:
Really? This "implicit return" was discussed here four weeks ago. It's current state and also described in the documentation. I don't understand what you want now?

Reply:
Pretend I review dozens of PRs here -- I lose track of what's actual and what's proposed, especially if I'm not deeply involved with that area of the code. I do recall discussing implicit return on one of your other PRs; was unaware it was implemented.

> It's current state and also described in the documentation.

Where? This in the documentation (L611):
> _Best practice_: To avoid confusion, do not execute further instructions in a test or hook after calling `this.skip()`.

That sentence _certainly_ doesn't imply an implicit return will occur. I find it horrid that `this.skip` does without using `return`.

Reply:
Climb one line up:
> It's also important to note that calling this.skip() will effectively abort the test.

If you are terrified by the "implicit return", act accordingly and open your PR.
It's current state and out of scope of this PR.

Reply:
Effectively aborting the test within a `before` to me only implies setting a flag, not implicit return (since the test has not yet begun execution).

As already stated, the implicit return behavior itself surprised me -- "terrified" is the wrong word -- "stunned" is more accurate. _Nowhere else_ in normal JavaScript coding (to my knowledge) would invoking a function cause implicit return from the caller's function.

Well aware it is outside scope of your PR here.

Reply:
Much of today's conversation has nothing to do with your PR. You do (to my knowledge) possess the most up-to-date knowledge on hook processing, so you're the person most likely to know the answers. Sorry if my questions came off as some sort of "last-minute-redo-this-PR" thing, as that was never my intention.